psp34_shiden/contracts/shiden34/src/lib.rs

```diff
@@ -98,34 +135,36 @@ pub mod shiden34 {
                 _instance._init_with_owner(caller);
             })
         }
+    }
 
-        /// Check if the transferred mint values is as expected
-        fn check_value(&self, transfered_value: u128, mint_amount: u64) -> Result<(), PSP34Error> {
-            if let Some(value) = (mint_amount as u128).checked_mul(self.price_per_mint) {
-                if transfered_value == value {
-                    return Ok(())
-                }
-            }
-
-            return Err(PSP34Error::Custom("BadMintValue".to_string()))
+    impl psp34::Internal for Shiden34Contract {
+        fn _emit_transfer_event(&self, from: Option<AccountId>, to: Option<AccountId>, id: Id) {
+            self.env().emit_event(Transfer { from, to, id });
         }
 
-        /// Check amount of tokens to be minted
-        fn check_amount(&self, mint_amount: u64) -> Result<(), PSP34Error> {
-            if mint_amount == 0 {
-                return Err(PSP34Error::Custom("CannotMintZeroTokens".to_string()))
-            }
-            if let Some(amount) = self.last_token_id.checked_add(mint_amount) {
-                if amount <= self.max_supply {
-                    return Ok(())
-                }
-            }
-            return Err(PSP34Error::Custom("CollectionFullOrLocked".to_string()))
+        fn _emit_approval_event(
+            &self,
+            from: AccountId,
+            to: AccountId,
+            id: Option<Id>,
+            approved: bool,
+        ) {
+            self.env().emit_event(Approval {
+                from,
+                to,
+                id,
+                approved,
+            });
         }
 
-        /// Check if token is minted
-        fn token_exists(&self, id: Id) -> Result<(), PSP34Error> {
-            self.owner_of(id).ok_or(PSP34Error::TokenNotExists)?;
+        fn _do_safe_transfer_check(
+            &mut self,
+            _operator: &AccountId,
+            _from: &AccountId,
+            _to: &AccountId,
+            _id: &Id,
+            _data: &Vec<u8>,
```

Comment:
ink_prelude::vec::Vec import is missing, so it cannot be compiled.
https://github.com/AstarNetwork/wasm-showcase-dapps/pull/27/files#diff-c084cdf45f42bdb6e563636cee056161da26af1be4bc7bde84451e27ad2836b4R11

Reply:
did you try to compile? it actually works for me.

Reply:
Yes, getting this error.
<img width="1511" alt="Screen Shot 2022-11-13 at 17 15 27" src="https://user-images.githubusercontent.com/28953860/201512525-72f6f56d-4d20-4c79-8903-df7a72f81bb4.png">